uwh-overlay-drawing/src/main.rs

```diff
@@ -0,0 +1,181 @@
+use std::{str::FromStr, sync::mpsc::channel};
+//use uwh_common::game_snapshot::GamePeriod;
+use network::State;
+use std::net::IpAddr;
+
+use macroquad::prelude::*;
+use uwh_common::game_snapshot::GamePeriod;
+mod load_images;
+mod network;
+mod pages_alpha;
+mod pages_color;
+
+fn window_conf() -> Conf {
+    Conf {
+        window_title: "UWH Overlay".to_owned(),
+        window_width: 1920,
+        window_height: 1080,
+        window_resizable: false,
+        ..Default::default()
+    }
+}
+
+pub struct AppConfig {
+    refbox_ip: IpAddr,
+    refbox_port: u64,
+    uwhscores_url: String,
+}
+
+impl AppConfig {
+    /// Get the app config's refbox ip.
+    pub fn refbox_ip(&self) -> IpAddr {
+        self.refbox_ip
+    }
+
+    /// Get the app config's refbox port.
+    pub fn refbox_port(&self) -> u64 {
+        self.refbox_port
+    }
+
+    /// Get a reference to the app config's uwhscores url.
+    pub fn uwhscores_url(&self) -> &str {
+        self.uwhscores_url.as_ref()
+    }
+}
+
+impl Default for AppConfig {
+    fn default() -> Self {
+        AppConfig {
+            refbox_ip: IpAddr::from_str("127.0.0.1").unwrap(),
+            refbox_port: 8000,
+            uwhscores_url: String::from("uwhscores.com"),
+        }
+    }
+}
+
+#[macroquad::main(window_conf)]
+async fn main() {
+    let (tx, rx) = channel::<State>();
+    let mut conf_directory = home::home_dir().unwrap();
+    conf_directory.push(".config/uwh-overlay/config.json");
+    let config = if let Ok(file) = std::fs::read(conf_directory) {
+        let data: serde_json::Value = serde_json::from_str(std::str::from_utf8(&file).unwrap())
+            .expect("Ill formatted config file!");
+        AppConfig {
+            refbox_ip: IpAddr::from_str(data["refbox_ip"].as_str().unwrap()).unwrap(),
+            refbox_port: data["refbox_port"].as_u64().unwrap(),
+            uwhscores_url: data["uwhscores_url"].as_str().unwrap().to_string(),
+        }
+    } else {
+        AppConfig::default()
+    };
+    std::thread::spawn(|| {
+        network::networking_thread(tx, config)
+            .expect("Networking error. Does the supplied URL exist and is it live?")
+    });
+    let args: Vec<String> = std::env::args().collect();
+    let mut animation_counter = 0f32;
+    let mut secondary_animation_counter = 0f32; // for when we call two draw functions with animations, we need two counters
+    assert!(
+        args.len() == 2,
+        "Got {} args instead of one. Pass one argument, --color or --alpha to get the color or alpha feed respectively",
+        args.len() - 1
+    );
+    let (textures, is_alpha_mode) = if args[1] == *"--color" {
+        (load_images::Textures::init_color(), false)
+    } else if args[1] == *"--alpha" {
+        (load_images::Textures::init_alpha(), true)
+    } else {
+        panic!("Expected --color or --alpha arg!")
+    };
```

Comment:
Reading CLI arguments is just a temporary thing until I merge the colour and alpha programs into one (detailed above)

Reply:
I suspect there will always be a need for at least some CLI arguments. For example, the final product will need to be fullscreen, but for dev, windows are easier. A `-f` flag would be an easy solution

Reply:
I personally use a window manager, so the flag isn't necessary for me, but if you think it would be helpful for you, or if you can think of some other CLI argument that would be useful in the future, I'll add it and move to `clap`.